File: csharp-api/Models/PreLoginRecommendation.cs
namespace ScreenGuardianAPI.Models
{
    public class PreLoginRecommendation
    {
        public int Id { get; set; }
        public int MinAge { get; set; }
        public int MaxAge { get; set; }
        public int RecommendedMinutes { get; set; }
        public string Message { get; set; }
    }
}

File: client/src/api/PreLoginRecommendation.ts
import { CSHARP_API_BASE_URL } from "../config/env";

export type ScreenTimeRecommendation = {
  id: number;
  minAge: number;
  maxAge: number;
  recommendedMinutes: number;
  message: string;
};

export async function getPreLoginRecommendation(age: number) {
  const res = await fetch(
    `${CSHARP_API_BASE_URL}/api/preloginrecommendation/recommendation?age=${age}`
  );

  if (!res.ok) throw new Error("Failed");

  return res.json();
}